src/brogue/Monsters.c

```diff
@@ -747,12 +752,94 @@ void drawManacles(short x, short y) {
     }
 }
 
+boolean getRandomHordeSpawnLocation(short loc[2], short hordeID, boolean periodic) {
+    boolean validTurretLoc;
+    short i, j, k, newX, newY;
+    creatureType* leaderType = &(monsterCatalog[hordeCatalog[hordeID].leaderType]);
+    unsigned long avoidedTerrainFlags = avoidedFlagsForMonster(leaderType);
+    unsigned long avoidedMapFlags = HAS_PLAYER | HAS_MONSTER | HAS_STAIRS | HAS_ITEM | IS_IN_MACHINE;
+
+    short **grid = allocGrid();
+    if (periodic) {
+        calculateDistances(grid, player.xLoc, player.yLoc, T_DIVIDES_LEVEL, NULL, true, true);
+        findReplaceGrid(grid, -30000, DCOLS/2-1, 0);
+        findReplaceGrid(grid, 30000, 30000, 0);
+        findReplaceGrid(grid, DCOLS/2, 30000-1, 1);
+    } else {
+        fillGrid(grid, 1);
+    }
+
+    if (periodic) {
+        avoidedMapFlags |= ANY_KIND_OF_VISIBLE;
+    } else {
+        // We need to exclude tiles marked with IN_FIELD_OF_VIEW to prevent the initial monsters from spawning within FOV of the entry stairs.
+        // IN_FIELD_OF_VIEW is specifically calculated from the entry stairs when the level is generated.
+        avoidedMapFlags |= IN_FIELD_OF_VIEW;
+    }
+
+    // Fiery monsters will not spawn in flammable terrain.
+    if (leaderType->flags & MONST_FIERY) {
+            avoidedTerrainFlags |= T_IS_FLAMMABLE;
+    }
+
+    getTerrainGrid(grid, 0, avoidedTerrainFlags, avoidedMapFlags);
+
+    for (i = 0; i < DCOLS; i++) {
+        for (j = 0; j < DROWS; j++) {
+            if (hordeCatalog[hordeID].spawnsIn && !cellHasTerrainType(i, j, hordeCatalog[hordeID].spawnsIn)) {
+                grid[i][j] = 0;
+            }
+
+            // This is the only check applied to periodic hordes.
+            if (periodic) {
+                continue;
+            }
```

Comment:
Is this to retain old behaviour, where periodic hordes selected only from `getRandomMonsterSpawnLocation`? Do you envisage any problems if we were to remove this? I suppose it would be just that periodic spawns can happen in fewer places

Reply:
Yup, just to retain old behavior. I don't think it would make a massive difference if removed.